### PR TITLE
Premint - better mint arguments.  Mint recipient is specified

### PR DESCRIPTION
**Pull request by oveddan**
Mint arguments are a struct, and now mint recipient can be specified

### CODE REVIEW

**Comment by oveddan** [comment]
Current dependencies on/for this PR:
* main
  * **PR #278** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/278" target="_blank">Graphite</a> 
    * **PR #283** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/283" target="_blank">Graphite</a> 
      * **PR #290** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/290" target="_blank">Graphite</a> 
        * **PR #291** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/291" target="_blank">Graphite</a> 
          * **PR #262** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/262" target="_blank">Graphite</a> 
            * **PR #294** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/294" target="_blank">Graphite</a>  👈
            * **PR #289** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/289" target="_blank">Graphite</a> 
            * **PR #273** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/273" target="_blank">Graphite</a> 
              * **PR #288** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/288" target="_blank">Graphite</a> 
                * **PR #284** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/284" target="_blank">Graphite</a> 
              * **PR #280** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/280" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ourzora/zora-protocol/294?utm_source=stack-comment).

**Comment by oveddan** [comment]
todo: 
[ ] if mintRecipient is not specified, default to `msg.sender`

File: packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol [hunk 1]
```diff
@@ -43,6 +43,12 @@ interface ILegacyZoraCreator1155PremintExecutor {
     ) external payable returns (uint256 newTokenId);
 }
 
+struct MintArguments {
```
Comment (iainnash) [inline packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol:46]: Natspec here for where these get passed?

MintRecipient and MintComment go to fixedpriceminter, while mintReferral is used in the 1155impl contract.

File: packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol [hunk 2]
```diff
@@ -151,7 +156,7 @@ contract ZoraCreator1155PremintExecutorImpl is
         uint256 quantityToMint,
         address fixedPriceMinter,
         uint32 uid,
-        bytes memory mintArguments
+        MintArguments memory mintArguments
```
Comment (iainnash) [inline packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol:159]: can this be calldata here?

**Review by iainnash — APPROVED** [review]

**Comment by oveddan** [comment]
merge into parent branch

**Pull request closed without merging**